File: backend/KayraExportTask2/KayraExportTask2Domain/Entities/Product.cs
using KayraExportTask2Domain.Entities.Common;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace KayraExportTask2Domain.Entities
{
    public class Product : BaseEntity
    {
        public string Name { get; set; }
        public string Description { get; set; }
        public decimal Price { get; set; }
        public int Stock {  get; set; }
        public string Slug { get; set; }
        public Guid CategoryId { get; set; }
        public Category Category { get; set; }
        public ICollection<ProductImage> Images { get; set; } = new List<ProductImage>();

    }
}

File: frontend/src/app/admin/models/Product.ts
import { ProductImage } from "./ProductImage";
export interface Product{
    id: string;
    name: string;
    description: string;
    price: number;
    stock: number;
    categoryId: string;
    productImages: ProductImage[];
    slug:string;
}

export interface CreateProduct {
  name: string;
  price: number;
  stock: number;
  categoryId: string;
  description: string;
  productImages: ProductImage[];  
  slug:string;
}
export interface UpdateProduct extends CreateProduct {
  id: string;
}